common/src/main/java/org/valkyrienskies/mod/mixin/feature/ai/node_evaluator/SwimNodeEvaluatorMixin.java

```diff
@@ -0,0 +1,188 @@
+            } else if (blockGetter instanceof Level) {
+                level = (Level) blockGetter;
+            }
+            if (level != null) {
+                final double origX = blockPos.getX();
+                final double origY = blockPos.getY();
+                final double origZ = blockPos.getZ();
+                final Level finalLevel = level;
+                VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                    origY, origZ, 1,
+                    (x, y, z) -> {
+                        final BlockPos groundPos = new BlockPos(x, y, z);
+                        isPathFindable[0] = instance.isPathfindable(finalLevel, groundPos, pathComputationType);
+                    });
+            }
+        }
+        return isPathFindable[0];
+    }
+    //endregion
+
+    //region Area obstacle path type
+    @Redirect(
+        at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/level/PathNavigationRegion;getFluidState(Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/material/FluidState;"),
+        method = "isFree"
+    )
+    private FluidState getFluidStateRedirectIsFree(final PathNavigationRegion instance, final BlockPos pos) {
+        final FluidState[] fluidState = {instance.getFluidState(pos)};
+        if (fluidState[0].isEmpty() && VSGameConfig.SERVER.getAiOnShips()) {
+            final Level level = ((PathNavigationRegionAccessor) instance).getLevel();
+
+            final double origX = pos.getX();
+            final double origY = pos.getY();
+            final double origZ = pos.getZ();
+            VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                origY, origZ, 1,
+                (x, y, z) -> {
+                    final BlockPos groundPos = new BlockPos(x, y, z);
+                    fluidState[0] = level.getFluidState(groundPos);
+                });
+        }
+        return fluidState[0];
+    }
+
+    @Redirect(
+        at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/level/PathNavigationRegion;getBlockState(Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/block/state/BlockState;"),
+        method = "isFree"
+    )
+    private BlockState getBlockStateRedirectIsFree(final PathNavigationRegion instance, final BlockPos pos) {
+        final BlockState[] blockState = {instance.getBlockState(pos)};
+        if (blockState[0].isAir() && VSGameConfig.SERVER.getAiOnShips()) {
+            final Level level = ((PathNavigationRegionAccessor) instance).getLevel();
+
+            final double origX = pos.getX();
+            final double origY = pos.getY();
+            final double origZ = pos.getZ();
+            VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                origY, origZ, 1,
+                (x, y, z) -> {
+                    final BlockPos groundPos = new BlockPos(x, y, z);
+                    blockState[0] = level.getBlockState(groundPos);
```

Comment:
Why are you using `transformToNearbyShipsAndWorld` if it only uses the last x,y,z that it gets and ignores the rest?